crates/polars-time/src/chunkedarray/rolling_window/dispatch.rs

```diff
@@ -47,9 +47,8 @@ where
 
     let arr = ca.downcast_iter().next().unwrap();
     // "5i" is a window size of 5, e.g. fixed
-    let arr = if options.window_size.parsed_int {
+    let arr = if options.by.is_none() {
```

Comment:
rather than branching based on `window_size.parsed_int` (which can lead to some surprising error messages), I think it's cleaner to branch on `by.is_none`